Add ability to paste images

nheagy opened the pull request:
This begins to address #655 by allowing images on the clipboard to be pasted into the editor. It's also the first step for https://github.com/wordpress-mobile/WordPress-iOS/issues/4815 (the second step being adding support to WP-iOS itself).

The general design is: if the paste results in an empty string in the JS, use the callback to have native code access the pasteboard. If this design is acceptable, it can be expanded in the future to add support for videos, multiple images, or a mix of text, images, and video.

/cc @SergioEstevao @diegoreymendez 

Thanks, and let me know what you think.

SergioEstevao commented on Assets/ZSSRichTextEditor.js:
```diff
@@ -107,7 +107,12 @@ ZSSEditor.init = function(callbacker, logger) {
         e.preventDefault();
         var plainText = (e.originalEvent || e).clipboardData.getData('text/plain');
-        document.execCommand('insertText', false, plainText);
+        if (plainText.length > 0) {
```
@nheagy what happens if the paste has images and text? will the plain text be length zero also?

nheagy replied:
If it has both, the text will paste and the image will not. IMO the best way to solve this is to have the native code trigger with all pastes, where it can handle multiple items on the pasteboard.

Not sure if that's something to implement now, or as a second phase after this one?

SergioEstevao replied:
It's ok for now to keep as it is, depending on the progress you make, we can come back here and handle the other scenarios.

diegoreymendez replied:
I think the scope of allowing single images to be pasted is fine, for starters.

diegoreymendez commented on Classes/WPEditorView.h:
```diff
@@ -155,6 +155,16 @@ stylesForCurrentSelection:(NSArray*)styles;
+ *
+ */
+- (void)editorView:(WPEditorView*)editorView
+     imagePasted:(UIImage *)image;
```
The alignment's a bit off in this line.

nheagy replied:
Should be fixed now.

diegoreymendez commented on Example/EditorDemo/WPViewController.m:
```diff
@@ -123,6 +123,31 @@ - (void)editorViewController:(WPEditorViewController *)editorViewController imag
+
+    [self.editorView insertLocalImage:[[NSURL fileURLWithPath:path] absoluteString] uniqueId:imageID];
+
+    NSProgress *progress = [[NSProgress alloc] initWithParent:nil userInfo:@{ @"imageID": imageID, @"url": path }];
```
Starting in this line, the code seems very similar to [this](https://github.com/wordpress-mobile/WordPress-Editor-iOS/blob/develop/Example/EditorDemo/WPViewController.m#L313).

I'd suggest adding a method to handle it instead.

nheagy replied:
Refactored a big chunk of both methods. Let me know what you think.

diegoreymendez commented:
@nheagy - Sorry for the delay in reviewing this.  It's looking good on my end.  All comments are minor things.

At some point this is going to help us expans our copy & paste support for more elaborate content, but for now, being able to paste single images is a nice addition.

nheagy commented:
@diegoreymendez @SergioEstevao Thanks for looking this over.

I've fixed those few things, now. I also changed from PNG to JPEG representation of the NSData to avoid problems with rotation. This only affects the demo app, the WordPress-iOS app (or any other app using the editor) will still get a UIImage from the editor itself, and can implement more robust rotation code.

diegoreymendez commented:
:shipit:
